lib/IDE/CodeCompletion.cpp

```diff
@@ -1790,15 +1773,34 @@ void CodeCompletionCallbacksImpl::doneParsing(SourceFile *SrcFile) {
       default:
         break;
       }
-    }
-    if (!ExpectedCustomAttributeKinds) {
+
+      switch (*AttTargetDK) {
+      case DeclKind::Var:
+      case DeclKind::Subscript:
+        ExpectedCustomAttributeKinds |= CustomAttributeKind::VarMacro;
+        break;
+      case DeclKind::Struct:
+      case DeclKind::Class:
+      case DeclKind::Protocol:
+      case DeclKind::Enum:
+        ExpectedCustomAttributeKinds |= CustomAttributeKind::ContextMacro;
+        break;
```

Comment:
How about `Protocol` and `Extension`? If `ContextMacro` don't support them, could you add a comment?

Reply:
Thanks. I don't see why they wouldn't, but I'll double check. I briefly thought about extensions in the middle of doing something else and then promptly forgot about it :).

Reply:
(note that this is just extensions, protocol is already listed there)